crates/mod-crypto/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use anyhow::anyhow;
+use aws_lc_rs::cipher::{
+    AES_128, AES_256, DecryptionContext, PaddedBlockDecryptingKey, PaddedBlockEncryptingKey,
+    UnboundCipherKey,
+};
+
+use data_encoding::{
+    BASE32, BASE32HEX, BASE32HEX_NOPAD, BASE32_NOPAD, BASE64, BASE64URL, BASE64URL_NOPAD,
+    BASE64_NOPAD, HEXLOWER,
+};
+use config::{any_err, from_lua_value, get_or_create_sub_module};
+use mlua::prelude::LuaUserData;
+use mlua::{Lua, MetaMethod, UserDataFields, UserDataMethods, Value, Error as LuaError};
+use data_loader::KeySource;
+use hex::decode;
+use std::str;
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct AesParams {
+    pub algorithm: AesAlgo,
+    pub key: AesKey,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum AesAlgo {
+    Ecb(),
+    Cbc(),
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct LuaCfg {
+    pub key: KeySource,
+    pub value: String,
+    pub algorithm: String,
```

Comment:
if we type `algorithm`, then the `Deserialize` logic will parse the enum name for us (especially together with the removal of the parens from the variants above).

It will be case sensitive, which is fine and is consistent with the other type handling throughout kumo.

```suggestion
    pub algorithm: AesAlgo,
```